src/fio/fioEngine.js

```diff
@@ -287,192 +264,220 @@ export class FioEngine extends CurrencyEngine {
           account_name: actor,
           pos: -1,
           offset: -1
         },
-        this.currencyInfo.defaultSettings.historyNodeActions.getActions
+        HISTORY_NODE_ACTIONS.getActions
       )
 
-      if (lastActionObject.actions && lastActionObject.actions.length) {
+      asHistoryResponse(lastActionObject)
+      if (lastActionObject.actions.length) {
+        asHistoryNodeAction(lastActionObject.actions[0])
         lastActionSeqNumber = lastActionObject.actions[0].account_action_seq
+      } else {
+        // if no transactions at all
+        return true
+      }
+      if (lastActionObject.error && lastActionObject.error.noNodeForIndex) {
+        return false
       }
     } catch (e) {
-      console.log(e)
-      this.log(e)
-    }
-
-    if (!lastActionSeqNumber) {
-      return true
+      return this.checkTransactions(++historyNodeIndex)
     }
 
-    const limit = 10
-    let offset = 0
-
-    // history node saves up to 1000 records, so if seq number is more than 1000 we set pos to smallest value
-    if (lastActionSeqNumber > 1000) {
-      offset = lastActionSeqNumber - 1000
-    }
-
-    while (true) {
-      this.log('looping through checkTransactions')
-      const actionsObject = await this.multicastServers(
-        'history',
-        {
-          account_name: actor,
-          pos: offset,
-          offset: limit
-        },
-        this.currencyInfo.defaultSettings.historyNodeActions.getActions
-      )
-      let actions = []
-      actionsObject.actions.sort((a, b) => a.block_num - b.block_num)
+    const offset = 20
+    let pos = lastActionSeqNumber
+    let finish = false
 
-      if (actionsObject.actions && actionsObject.actions.length > 0) {
-        actions = actionsObject.actions
-      } else {
+    while (!finish) {
+      if (pos < 0) {
         break
       }
+      let actionsObject
+      try {
+        actionsObject = await this.requestHistory(
+          historyNodeIndex,
+          {
+            account_name: actor,
+            pos,
+            offset: -offset
+          },
+          HISTORY_NODE_ACTIONS.getActions
+        )
+        if (actionsObject.error && actionsObject.error.noNodeForIndex) {
+          return false
+        }
+        actionsObject.actions.sort(
+          (a, b) => b.account_action_seq - a.account_action_seq
+        )
 
-      for (let i = 0; i < actions.length; i++) {
-        const action = actions[i]
-        const blockNum = this.processTransaction(action, actor)
+        let actions = []
 
-        if (blockNum > newHighestTxHeight) {
-          newHighestTxHeight = blockNum
+        if (actionsObject.actions && actionsObject.actions.length > 0) {
+          actions = actionsObject.actions
+        } else {
+          break
         }
-      }
 
-      if (!actions.length || actions.length < limit) {
-        break
+        for (let i = 0; i < actions.length; i++) {
+          const action = actions[i]
+          asHistoryNodeAction(action)
```

Comment:
confirm that all history nodes return me actions from oldest to newest in one chunk